editor/src/document/document_message_handler.rs

```diff
@@ -164,8 +164,18 @@ impl MessageHandler<DocumentsMessage, &InputPreprocessor> for DocumentsMessageHa
 					_ => format!("Untitled Document {}", new_doc_title_num),
 				};
 
+				let new_document = match serialized_contents {
+					None => DocumentMessageHandler::with_name(name),
+					Some(content) => {
+						let res = DocumentMessageHandler::with_name_content(name, content);
```

Comment:
I feel like the name should default to the name of the file you opened and not just untitled document.
I would prefer to just add an extra message  e.g. something like `DocumentMessage::OpenFile{name: String, contents:String}`. IIRC the NewDocument handler is pretty huge and could use a bit of a modularization anyway.

Reply:
Kept the same function but pulled out some logic and added another optional arg.